src/snowflake/snowpark/dataframe.py

```diff
@@ -4332,19 +4358,31 @@ def rename(
         rename_map = {k: v for k, v in zip(normalized_name_list, rename_list)}
         rename_plan = Rename(rename_map, self._plan)
 
+        # AST
+        for col, new_name in col_or_mapper.items():
+            kv_tuple_ast = expr.col_or_mapper.seq_map_val.kvs.add()
+            build_expr_from_snowpark_column_or_python_val(kv_tuple_ast.vs.add(), col)
```

Comment:
The order of arguments seems to be inconsistent between these 2 fns.

Reply:
Yeah, I believe the `build_expr_from_python_val` should have the `expr_builder` argument come first as well (this feels more intuitive and all of the other helper methods also have the arguments with the `proto.Expr` arg come first).

I can make a cleanup TODO or complete the change in this PR if the above sounds good. Its been irking me for a while :)